test/e2e/dra/dra.go

```
@@ -95,6 +94,21 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 				framework.Fail("NodePrepareResource should have been called again")
 			}
 		})
+
+		ginkgo.It("must reject pods which are not authorized to use a claim", func() {
+			parameters := b.resourceClaimParameters()
+			claim := b.externalClaim(v1.AllocationModeImmediate)
+			pod := b.podExternal()
+
+			// This bypasses scheduling and therefore the pod gets
+			// to run on the node although it never gets added to
+			// the `ReservedFor` field of the claim.
+			pod.Spec.NodeName = nodes.NodeNames[0]
+
+			b.create(ctx, parameters, claim, pod)
+			err := e2epod.WaitForPodFailedReason(f.ClientSet, pod, "xyz", time.Minute)
```

"xyz" must be replaced with the actual reason, once there is one.

I think kubelet should actually fail the pod instead of retrying, because such a situation should not have been allowed to occur in the first place. That makes `WaitForPodFailedReason` a suitable check. Otherwise we would have to check for "Pending" and certain error events.

---

We can make this test fail faster by letting the pod run to completion instead of keeping it running. Right now, `WaitForPodFailedReason` times out. If the pod completed, it should return as soon as it notices that.

---

@pohly this should be fixed by #34. PTAL.

---

Please take this test into your PR, then we can close this one here.

---

@bart0sh you didn't include the test in your PR, did you?

The new test therefore still needs to be merged into my branch. I can rebase it.

---

It was merged, just with a slightly different name.